### ISSUE TITLE
Add KeycloakTestResourceLifecycleManager 

### ISSUE THREAD

**Issue by sberyozkin** [quarkusio/quarkus#13957]
**Description**
In many OIDC related tests we have to add `QuarkusTestResourceLifecycleManager` implementations which setup Keycloak - realm, client, users.
@stuartwdouglas has proposed that we provide the one which can be reused in tests, which is a nice idea.

**Implementation ideas**
Users who would like to test `quarkus-oidc` against Keycloak should be able to type something like this:
```
@QuarkusTest
@QuarkusTestResource(TestKeycloakRealmResourceManager.class)
public class CodeFlowTest {

    @Test
    public void testCodeFlow() throws IOException {
    }
}
``` 

where `TestKeycloakRealmResourceManager.class` would by default set up `quarkus` realm, `quarkus-app` client, users `alice` as `admin` and `bob` as `user`, something like that, and possibly be configurable

CC @stuartwdouglas @pedroigor 

@cemnura let me know please if it can be of interest once the wiremock issues are complete :-)

**Comment by cemnura** [comment]
@sberyozkin I would love to give it a crack. I think we are close to wrapping up the wiremock issues or opaque token.

This would make testing quarkus services that use oidc with keycloak a breeze :)

**Comment by gsmet** [comment]
It should be called `KeycloakTestResourceLifecycleManager` btw.

**Comment by sberyozkin** [comment]
Hi @cemnura, OK, great, let us complete the wiremock issue first (I'll follow up with the docs) and then we'll discuss this one, cheers

**Comment by sberyozkin** [comment]
@cemnura Hi Cem - assigning to you as agreed, lets have a `keycloak-server` test module similar to a more generic wiremock based `oidc-server` - but this one would setup the real live KC instance, moving the code from `integration-tests/oidc`, thanks

**Comment by cemnura** [comment]
@sberyozkin I'm on it 😄

**Comment by cemnura** [comment]
I figured out that the current `integration-tests/oidc` runs a test container via maven plugin.

I migrated the `QuarkusTestResourceLifecycleManager` to a separate `test-framework` module and enabled the test container to run programmatically rather then a maven plugin.

All previous test seems to be working.

Also, I did not change the maven profile configurations so that it wont be a problem during CI pipeline.

To run the test 

```
./mvnw -Dquickly
./mvnw package -f integration-tests/oidc -Ptest-keycloak             
```
